src/NServiceBus.Core/Callbacks/RequestResponseExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿namespace NServiceBus
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Callbacks;
+
+    /// <summary>
+    /// 
+    /// </summary>
+    public static class RequestResponseExtensions
+    {
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <typeparam name="TResponse"></typeparam>
+        /// <param name="bus"></param>
+        /// <param name="requestMessage"></param>
+        /// <returns></returns>
+        public static SendContext<TResponse> RequestResponse<TResponse>(this IBus bus, object requestMessage)
+        {
+            Guard.AgainstNull(requestMessage, "requestMessage");
+            Guard.AgainstNull(bus, "bus");
+
+            var correlationId = Guid.NewGuid().ToString();
+
+            bus.SetMessageHeader(requestMessage, "NServiceBus.HasCallback", Boolean.TrueString);
```

Comment:
I don't like 'HasCallback' name but that's an impl detail. I think we should figure out something more meaningful for the receiver.